src/prototype_creep_resources.js

```diff
@@ -297,7 +303,8 @@ Creep.prototype.transferToCreep = function(direction) {
   const creeps = adjacentPos.lookFor('creep');
   for (let i = 0; i < creeps.length; i++) {
     const otherCreep = creeps[i];
-    if (!checkCreepForTransfer(otherCreep)) {
+    // don't transfer to other base's creep, fix carry helper
+    if (!checkCreepForTransfer(otherCreep) || otherCreep.memory.base !== this.memory.base) {
```

Comment:
I don't think this is the best approach. Maybe fixes the current issue.

I had a couple of cases where the external rooms of different base rooms are crossing each other and it doesn't matter to which room the energy is transferred.

I think the better approach would be checking for the direction. `helper` is the newest sort of carry usage, maybe we should rethink that, add a value to the memory to be able to check for them, have a different role type, ...

Reply:
What about `this.memory.helper && !otherCreep.memory.helper && otherCreep.memory.base !== this.memory.base`?

In fact, I disabled transfer to carry in PR #403. I don't think it necessary to allow energy transfer between carries because it will mess up carry's routing.

Reply:
I think this approach is better. We can also restrict, that helpers can only transfer to helpers, carries only to carries that should be fine.

Not sure if I understand your last sentence

Reply:
check out the new commit plz :)

I'm sorry for my poor English if my last sentence confuse you...  I found that carries near `pathStart` will get energy from other carries and not move to their own target.  This problem is obvious in low RCL rooms.  I just forbade energy transfer between carries in my scripts so that every carry can reach their targets.